src/features/collectables/HotspotListItem.tsx

```diff
@@ -1,42 +1,83 @@
-import React, { useMemo } from 'react'
+import React, { useEffect, useMemo } from 'react'
 import { FadeIn, FadeOut } from 'react-native-reanimated'
 import { PublicKey } from '@solana/web3.js'
+import { BoxProps } from '@shopify/restyle'
 import Text from '../../components/Text'
 import TouchableOpacityBox from '../../components/TouchableOpacityBox'
-import { Collectable } from '../../types/solana'
+import { CompressedNFT } from '../../types/solana'
 import { ww } from '../../utils/layout'
 import { useHotspot } from '../../hooks/useHotspot'
 import Box from '../../components/Box'
 import MobileSymbol from '../../assets/images/mobileSymbol.svg'
 import { removeDashAndCapitalize } from '../../utils/hotspotNftsUtils'
 import { ReAnimatedBox } from '../../components/AnimatedBox'
 import ImageBox from '../../components/ImageBox'
+import { Theme } from '../../theme/theme'
+import IotSymbol from '../../assets/images/iotSymbol.svg'
+import { hotspots } from '../../store/slices/hotspotsSlice'
+import { useAppDispatch } from '../../store/store'
+import { useAccountStorage } from '../../storage/AccountStorageProvider'
 
 export type HotspotListItemProps = {
-  hotspot: Collectable
-  onPress: (hotspot: Collectable) => void
-}
+  hotspot: CompressedNFT
+  onPress: (hotspot: CompressedNFT) => void
+} & BoxProps<Theme>
 
-const HotspotListItem = ({ hotspot, onPress }: HotspotListItemProps) => {
+const HotspotListItem = ({
+  hotspot,
+  onPress,
+  ...rest
+}: HotspotListItemProps) => {
   const COLLECTABLE_HEIGHT = ww / 2
-  const { json } = hotspot
-  const mint = useMemo(
-    () => new PublicKey(hotspot.mint.address),
-    [hotspot.mint],
+  const dispatch = useAppDispatch()
+  const { currentAccount } = useAccountStorage()
+  const {
+    content: { metadata },
+  } = hotspot
+  const mint = useMemo(() => new PublicKey(hotspot.id), [hotspot.id])
+  const { pendingMobileRewards, pendingIotRewards } = useHotspot(mint)
+
+  const hasIotRewards = useMemo(
+    () => pendingIotRewards && pendingIotRewards > 0,
+    [pendingIotRewards],
   )
-  const { pendingRewards } = useHotspot(mint)
 
-  // TODO: Add IOT Rewards once IOT MINT is available
   const hasMobileRewards = useMemo(
-    () => pendingRewards && pendingRewards > 0,
-    [pendingRewards],
+    () => pendingMobileRewards && pendingMobileRewards > 0,
+    [pendingMobileRewards],
   )
 
+  useEffect(() => {
+    if (!currentAccount) return
+
+    dispatch(
```

Comment:
This will dispatch updates whenever anything changes in any of the list items? Is that intended or is it supposed to just be called initially?

Reply:
Hmm let me look into this. We wanna update when it first loads so maybe I can do that for now

Reply:
Fixed